.changeset/proud-seals-complain.md

```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': patch
+---
+
+Use immutable beacon address in BeaconProxy
```

Comment:
```suggestion
`BeaconProxy`: Use an immutable variable to store the address of the beacon. It is no longer possible for a `BeaconProxy` to upgrade by changing to another beacon.
```

Reply:
> It is no longer possible for a `BeaconProxy` to upgrade by changing to another beacon.

From what I understand, this was not possible in the first place, unless users extended the BeaconProxy to allow this. We never exposed a way to modify the beacon storage slot in the BeaconProxy.

Reply:
You could do it if there is a function in the implementation that sets the ERC1967 slot for beacon